### PR TITLE
Add overwriting scene entities given a scene instance

### PR DESCRIPTION
**Pull request by dmyyy**
# Objective

Implements #13659 

## Solution

- Add `overwrite_in_world_for_instance` to `Scene`
- Add `WorldId` to `InstanceInfo`

Please take a look at unit test for example usage

### CODE REVIEW

File: crates/bevy_scene/src/scene.rs [hunk 1]
```diff
@@ -47,7 +50,9 @@ impl Scene {
         Ok(Self { world: new_world })
     }
 
-    /// Write the entities and their corresponding components to the given world.
+    /// Writes all resources/entities and their corresponding components in the scene world to new
+    /// entities in the given world. On success returns an instance mapping entities in the scene world
```
Comment (alice-i-cecile) [inline crates/bevy_scene/src/scene.rs:54]: Style: the second sentence should have an empty line, and then be on its own line :)

**Comment by alice-i-cecile** [comment]
I think I'm starting to understand the use of this feature, but the docs / naming / motivation needs more work. Ping me after the 0.14 release and I can spend some time helping you put this into words if you're feeling stuck :)

**Comment by dmyyy** [comment]
> I think I'm starting to understand the use of this feature, but the docs / naming / motivation needs more work. Ping me after the 0.14 release and I can spend some time helping you put this into words if you're feeling stuck :)

Sounds good - wanted to get a little bit of buy-in before I designed the feature (beyond what I directly need). Will do a second pass and ping some time after 0.14.

**Comment by dmyyy** [comment]
Closing in favor of: https://github.com/bevyengine/bevy/pull/13714

**Pull request closed without merging**